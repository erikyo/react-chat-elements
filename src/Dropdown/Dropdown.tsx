import type React from "react";
import { useState } from "react";
import "./Dropdown.css";
import classNames from "classnames";
import Button from "../Button/Button";
import type { IDropdownProps, IDropdownItemType } from "../type";

const Dropdown = ({
	animationPosition = "nortwest",
	animationType = "default",
	...props
}: IDropdownProps): JSX.Element => {
	const [show, setShow] = useState<boolean | undefined>(undefined);

	const onBlur = () => {
		if (show === true) setShow(false);
	};

	return (
		<div
			className={classNames("rce-dropdown-container", props.className)}
			onBlur={onBlur}
		>
			{<Button {...props.buttonProps} onClick={() => setShow(!show)} />}
			<div
				className={classNames(
					"rce-dropdown",
					animationType,
					`rce-dropdown-open__${animationPosition}`,
					{ "dropdown-hide": show === false },
					{ "dropdown-show": show === true },
				)}
			>
				<ul>
					{props.title && (
						<span className="rce-dropdown-title">{props.title}</span>
					)}
					{props.items?.map((x: IDropdownItemType, i) => (
						<li
							key={`dropdown-item-${i.toString()}`}
							onMouseDown={(e) => props?.onSelect?.(e, i)}
						>
							{typeof x !== "string" ? (
								x.icon ? (
									<span className="rce-button-icon--container">
										{(x.icon.float === "right" || !x.icon.float) && (
											<a href={"#"}>{x.text}</a>
										)}

										<span
											style={{
												float: x.icon.float,
												color: x.icon.color,
												fontSize: x.icon.size || 12,
											}}
											className={classNames(
												"rce-button-icon",
												x.icon.className,
											)}
										>
											{x.icon.component}
										</span>

										{x.icon.float === "left" && <a href={"#"}>{x.text}</a>}
									</span>
								) : (
									<a href={"#"}>{x.text}</a>
								)
							) : (
								<a href={"#"}>{x}</a>
							)}
						</li>
					))}
				</ul>
			</div>
		</div>
	);
};

export default Dropdown;
