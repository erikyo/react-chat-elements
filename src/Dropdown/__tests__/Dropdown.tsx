import React, { Component } from "react";
import { shallow } from "enzyme";
import { describe, expect, it } from 'vitest'
import toJson from "enzyme-to-json";
import Dropdown from "../Dropdown";

describe("Dropdown component", () => {
	it("should render without issues", () => {
		const component = shallow(<Dropdown />);
		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});
});
