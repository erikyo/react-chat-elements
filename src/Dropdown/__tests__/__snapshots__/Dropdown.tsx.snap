// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > should render without issues 1`] = `
<div>
  <div
    class="rce-dropdown-container"
  >
    <button
      class="rce-button"
      style="background-color: rgb(57, 121, 170); color: white; border-color: #3979aa;"
      type="button"
    >
      <span />
    </button>
    <div
      class="rce-dropdown default rce-dropdown-open__nortwest"
    >
      <ul />
    </div>
  </div>
</div>
`;
